llm/llama.cpp/gen_linux.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# This script is intended to run inside the go generate
+# working directory must be llm/llama.cpp
+
+# First we build our default built-in library which will be linked into the CGO
+# binary as a normal dependency. This default build is CPU based.
+#
+# Then we build a CUDA dynamic library (although statically linked with the CUDA
+# library dependencies for maximum portability)
+#
+# Then if we detect ROCm, we build a dynamically loaded ROCm lib.  ROCm is particularly
+# important to be a dynamic lib even if it's the only GPU library detected because
+# we can't redistribute the objectfiles but must rely on dynamic libraries at
+# runtime, which could lead the server not to start if not present.
+
+set -ex
+set -o pipefail
+
+echo "Starting linux generate script"
+if [ -z "${CUDACXX}" -a -x /usr/local/cuda/bin/nvcc ]; then
+    export CUDACXX=/usr/local/cuda/bin/nvcc
+fi
+COMMON_CMAKE_DEFS="-DCMAKE_POSITION_INDEPENDENT_CODE=on -DLLAMA_ACCELERATE=on -DLLAMA_NATIVE=off -DLLAMA_AVX=on -DLLAMA_AVX2=off -DLLAMA_AVX512=off -DLLAMA_FMA=off -DLLAMA_F16C=off"
+OLLAMA_DYN_LIB_DIR="gguf/build/lib"
+mkdir -p ${OLLAMA_DYN_LIB_DIR}
+touch ${OLLAMA_DYN_LIB_DIR}/.generated
+source $(dirname $0)/gen_common.sh
+init_vars
+git_module_setup
```

Comment:
this `git_module_setup` step will have to be done before creating the `OLLAMA_DYN_LIB_DIR` or the submodule clone will fail on the first generation due to the directory not being empty:
```
go generate ./...
+ set -o pipefail
+ echo 'Starting linux generate script'
Starting linux generate script
+ '[' -z '' -a -x /usr/local/cuda/bin/nvcc ']'
+ COMMON_CMAKE_DEFS='-DCMAKE_POSITION_INDEPENDENT_CODE=on -DLLAMA_ACCELERATE=on -DLLAMA_NATIVE=off -DLLAMA_AVX=on -DLLAMA_AVX2=off -DLLAMA_AVX512=off -DLLAMA_FMA=off -DLLAMA_F16C=off'
+ OLLAMA_DYN_LIB_DIR=gguf/build/lib
+ mkdir -p gguf/build/lib
+ touch gguf/build/lib/.generated
++ dirname ./gen_linux.sh
+ source ./gen_common.sh
+ init_vars
+ LLAMACPP_DIR=gguf
+ PATCHES=0001-Expose-callable-API-for-server.patch
+ CMAKE_DEFS=-DLLAMA_ACCELERATE=on
+ CMAKE_TARGETS='--target ggml --target ggml_static --target llama --target build_info --target common --target ext_server --target llava_static'
+ echo ''
+ grep -- -g
+ CMAKE_DEFS='-DCMAKE_BUILD_TYPE=Release -DLLAMA_ACCELERATE=on'
+ git_module_setup
+ '[' -n '' ']'
+ git submodule init
+ git submodule update --force gguf
fatal: destination path '/home/bruce/dhiltgen/ollama/llm/llama.cpp/gguf' already exists and is not an empty directory.
fatal: clone of 'https://github.com/ggerganov/llama.cpp.git' into submodule path '/home/bruce/dhiltgen/ollama/llm/llama.cpp/gguf' failed
Failed to clone 'llm/llama.cpp/gguf'. Retry scheduled
fatal: destination path '/home/bruce/dhiltgen/ollama/llm/llama.cpp/gguf' already exists and is not an empty directory.
fatal: clone of 'https://github.com/ggerganov/llama.cpp.git' into submodule path '/home/bruce/dhiltgen/ollama/llm/llama.cpp/gguf' failed
Failed to clone 'llm/llama.cpp/gguf' a second time, aborting
llm/llama.cpp/generate_linux.go:3: running "bash": exit status 1
```